src/VoteDelegateFactory.sol

```diff
@@ -21,7 +21,8 @@ import {VoteDelegate} from "src/VoteDelegate.sol";
 contract VoteDelegateFactory {
     // --- storage variables ---
 
-    mapping(address => uint256) public created;
+    mapping(address usr => address voteDelegate)     public delegates;
+    mapping(address voteDelegate => uint256 created) public created;
```

Comment:
I think we could avoid using two mappings by replacing the `created` mapping with a view function:
```
function created(address voteDelegate) public view returns (uint256) {
    (bool success, bytes memory res) = voteDelegate.staticcall(abi.encodeCall(VoteDelegate(voteDelegate).delegate, ()));
    return (success && res.length == 32 && delegates[abi.decode(res, (address))] == voteDelegate) ? 1 : 0;
}
 ```

Reply:
Probably a bit overkill but if we also want to protect against future precompiles that might allow returning arbitrary data, we could consider also adding the code check suggested in the past by Cantina:

```
function created(address voteDelegate) public view returns (uint256 c) {
    if (voteDelegate.code.length > 0) {
        (bool success, bytes memory res) = voteDelegate.staticcall(abi.encodeCall(VoteDelegate(voteDelegate).delegate, ()));
        c = (success && res.length == 32 && delegates[abi.decode(res, (address))] == voteDelegate) ? 1 : 0;
    }
}
```

Reply:
On the other hand, the view function adds ~4k gas to the `LockstakeEngine.selectVoteDelegate()` call. This being a more critical call to optimize than `VoteDelegateFactory.create()`, that would be an argument to keep the two mappings.

Reply:
> Ah that's a nice trick. Not sure it's too standard though, and honestly the amount of delegates that are being created is not that large. So maybe better to keep it simple.

Ok let's keep it as is, resolving.